grommet/grommet
# Adjust onSearch type definition

#### What does this PR do?
This PR adjusts the onSearch type definition for Select and DataTable to be: `(search: string) => void`

#### Where should the reviewer start?
src/js/components/Select/index.d.ts or src/js/components/DataTable/index.d.ts

#### What testing has been done on this PR?
This PR has been tested on a local TS project where the storybook example of Select Search was copied into my local TS project. The same was done for DataTable with searchable columns. There were no flags in the editor, and the application worked as expected. 

#### How should this be manually tested?
In a local TS project, copy and paste the code from the Select Search storybook example and make sure you are using a grommet build that contains this TS change. Do the same for the Tunable DataTable. There should be no flags in your code and the application should compile successfully.

#### Any background context you want to provide?
I found the information about what was being accepted as a parameter for onSearch in the SelectContainer.js file:
```
onSearch = debounce(search => {
    const { onSearch } = this.props;
    onSearch(search);
  }, debounceDelay(this.props));
```
I found for DataTable:
```
if (onSearch) {
      onSearch(nextFilters);
    }
```
I could see that it was accepting a search term variable. When I printed the `search` or `nextFilters` to the console, I saw that it was the string containing the search terms.

#### What are the relevant issues?
#3165 

#### Screenshots (if appropriate)

#### Do the grommet docs need to be updated?
No.

#### Should this PR be mentioned in the release notes?
No.

#### Is this change backwards compatible or is it a breaking change?
Yes, this is backwards compatible.
